src/util/integerfactoring.c

```diff
@@ -698,22 +1253,145 @@ static uint64_t gaIFLGetSmallestFactorv(int n, const ga_factor_list* fl, int* id
 	return hasFactors ? f : 1;
 }
 
+int      gaIFLsprintf(char* str, const ga_factor_list* fl){
```

Comment:
@abergeron A factor list constructed by `gaIFactorize()` can _always_ be printed in 128 bytes or less (corresponding to `2**64`). If you're printing something simple (error message or suchlike), and not constructing a huge string with `strb`, it's handy to just dump the factor list into some temporary `char buf[128]` on the stack using `gaIFLsprintf()` and output it as `%s` (string) using any `printf()`-like function. It requires no dynamic memory allocation.